plutus-core/budgeting-bench/Bench.hs

```diff
@@ -33,7 +34,7 @@ runTermBench name term = env
     (\_ -> bench name $ nf (unsafeEvaluateCek mempty defaultCostModel) term)
 
 benchSameTwoByteStrings :: StaticBuiltinName -> Benchmark
-benchSameTwoByteStrings name = createTwoTermBuiltinBench name (byteStringsToBench seedA) (byteStringsToBench seedA)
+benchSameTwoByteStrings name = createTwoTermBuiltinBench name (byteStringsToBench seedA) ((\(bs, e) -> (BS.copy bs, e)) <$> byteStringsToBench seedA)
```

Comment:
This seems mysterious but significant. Comment?

Reply:
I think @reactormonk mentioned an issue about this code in the channel. In the original code, probably GHC did common subexpression elimination on `(byteStringsToBench seedA)` which resulted the two bytestrings being the same array at the heap. This means that equality of these two bytestrings became O(1) pointer equality , instead of the expected O(n) element-wise equality check.

Reply:
A comment is definitely needed here in the code.

Reply:
Added the comment.